Apps/LabRecorder/CMakeLists.txt

```diff
@@ -1,223 +1,87 @@
-    IF (NOT BOOST_ROOT)
-        SET(BOOST_ROOT "C:/local/boost_1_63_0/")
-        SET(BOOST_LIBRARYDIR "C:/local/boost_1_63_0/lib64-msvc-14.0/")
-    ENDIF()
-ENDIF()
-SET(Boost_DEBUG                OFF) #Switch this and next to ON for help debugging Boost problems.
-SET(Boost_DETAILED_FAILURE_MSG OFF)
-set(Boost_USE_STATIC_LIBS      ON)
-set(Boost_USE_MULTITHREADED    ON)
-set(Boost_USE_STATIC_RUNTIME   OFF)
-IF(MSVC)
-    find_package(Boost REQUIRED COMPONENTS filesystem system thread chrono system date_time atomic iostreams regex zlib)
-    # Disable asio auto linking in date-time and regex
-    add_definitions(-DBOOST_DATE_TIME_NO_LIB)
-    add_definitions(-DBOOST_REGEX_NO_LIB)
-    set(boost_plus_libs ${Boost_LIBRARIES})
-ELSE()
-    find_package(Boost REQUIRED COMPONENTS filesystem system thread chrono system date_time atomic iostreams regex)
-    find_package(ZLIB)
-    set(boost_plus_libs ${Boost_LIBRARIES} ${ZLIB_LIBRARIES})
+	# Override by adding "-DBOOST_ROOT=C:\path\to\boost\ -DBOOST_LIBRARYDIR=C:\path\to\boost\lib32-msvc-14.0\" to your cmake command
+	IF (NOT BOOST_ROOT)
+	    SET(BOOST_ROOT "C:/local/boost_1_63_0/")
+		SET(BOOST_LIBRARYDIR "C:/local/boost_1_63_0/lib64-msvc-14.0/")
+		ENDIF()
 ENDIF()
 
-# LSL
-set(LSL_ROOT_DIR ${CMAKE_CURRENT_LIST_DIR}/../../LSL/liblsl)
-find_package(LSL REQUIRED)
-
- 
-###############
 # APPLICATION #
-###############
 
 # Target name
 set(target LabRecorder)
 
-# Sources
-set(sources
-    main.cpp
-    mainwindow.cpp
-    mainwindow.h
-    mainwindow.ui
-    recording.h
-    targetver.h
-)
-
 # Build executable
 add_executable(${target}
-    MACOSX_BUNDLE
-    ${sources}
+	MACOSX_BUNDLE
+	WIN32
+	main.cpp
+	mainwindow.cpp
+	mainwindow.h
+	mainwindow.ui
+	recording.h
+	#RecorderLib.h
+	#RecorderLib.cpp
+	targetver.h
 )
 
-# Project options
-set_target_properties(${target}
-    PROPERTIES
-    FOLDER "${IDE_FOLDER}"
-)
-
-target_include_directories(${target}
-    PRIVATE
-    ${Boost_INCLUDE_DIRS}
-    ${LSL_INCLUDE_DIRS})
+IF(MSVC)
+    find_package(Boost REQUIRED COMPONENTS filesystem iostreams thread zlib)
+	# Disable asio auto linking in date-time and regex
+	target_compile_definitions(${target} PRIVATE BOOST_DATE_TIME_NO_LIB BOOST_REGEX_NO_LIB)
+	target_link_libraries(${target} PRIVATE Boost::zlib)
+ELSE()
+    find_package(Boost REQUIRED COMPONENTS filesystem iostreams thread)
+	find_package(ZLIB REQUIRED)
+	target_link_libraries(${target} PRIVATE ${ZLIB_LIBRARIES})
+ENDIF()
 
 target_link_libraries(${target}
-    PRIVATE
-    Qt5::Core
-    Qt5::Gui
-    Qt5::Widgets
-    Qt5::Network
-    ${boost_plus_libs}
-    ${LSL_LIBRARIES}
+	PRIVATE
+	Qt5::Widgets
+	Qt5::Network
+	Boost::filesystem
+	Boost::iostreams
+	Boost::thread
+	LSL::lsl
 )
 
 # Fix a bug where building debug tries to link with non-debug lib.
-if(MSVC)
-    SET_TARGET_PROPERTIES(${target}
-        PROPERTIES
-            LINK_FLAGS "/NODEFAULTLIB:\"liblsl64\""
```

Comment:
I think this is necessary because of the #pragmas in `lsl_c.h`. We should start there